cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/SourceControlManagementService.java

```diff
@@ -230,7 +235,15 @@ public ApplicationRecord pullAndDeploy(ApplicationReference appRef) throws Excep
     SourceControlMeta sourceControlMeta = new SourceControlMeta(pullResponse.getApplicationFileHash());
     // Deploy with a generated uuid
     String versionId = RunIds.generate().getId();
-    ApplicationWithPrograms app = appLifecycleService.deployApp(appRef.app(versionId), appRequest,
+    ApplicationId appId = appRef.app(versionId);
+    
+    accessEnforcer.enforce(appId, authenticationContext.getPrincipal(), StandardPermission.CREATE);
```

Comment:
This API can also be called on pipeline as well, right? In that case we need a way to check whether the pipeline exists already or not, and check UPDATE vs CREATE accordingly.